test/test_autograd.py

```diff
@@ -1239,41 +1250,52 @@ def unpack_variables(args):
 
 for test in function_tests:
     cls, constructor_args, call_args = test[:3]
-    test_name = 'test_{}Function'.format(cls.__name__)
-    if len(test) == 4:
-        test_name += '_' + test[3]
-
-    def do_test(self, cls=cls, constructor_args=constructor_args,
-                call_args=call_args, test_name=test_name):
-        input = create_input(call_args)
-        self.assertEqual(gradcheck(cls(*constructor_args), input, eps=1e-6, atol=PRECISION), True)
-
-        if test_name not in ignore_inplace and issubclass(cls, InplaceFunction):
-            output = cls(*constructor_args)(*input)
-            if not isinstance(output, tuple):
-                output = (output,)
-            inplace_input = deepcopy(input)
-            inplace_input_copy = tuple(i + 0 for i in inplace_input)
-            fn = cls(*constructor_args, inplace=True)
-            inplace_output = fn(*inplace_input_copy)
-            if not isinstance(inplace_output, tuple):
-                inplace_output = (inplace_output,)
-            self.assertEqual(inplace_output, output)
-            # Check that gradient is the same
-            for inp_i, i in zip(inplace_input, input):
-                if inp_i.grad is not None:
-                    inp_i.grad.data.zero_()
-                if i.grad is not None:
-                    i.grad.data.zero_()
-            for io, o in zip(inplace_output, output):
-                grad = torch.randn(*io.size()).double()
-                io.backward(grad)
-                o.backward(grad)
-            for inp_i, i in zip(inplace_input, input):
-                self.assertEqual(inp_i.grad, i.grad)
-
-    assert not hasattr(TestAutograd, test_name), 'Two tests have the same name: ' + test_name
-    setattr(TestAutograd, test_name, do_test)
+    basic_test_name = 'test_{}Function'.format(cls.__name__)
+    if len(test) >= 4:
+        basic_test_name += '_' + test[3]
+
+    dim_args_idx = test[4] if len(test) == 5 else []
+
+    for dim_perm in product([-1, 1], repeat=len(dim_args_idx)):
+        test_name = basic_test_name
+        new_constructor_args = list(constructor_args)
+        for i, arg_idx in enumerate(dim_args_idx):
+            new_constructor_args[arg_idx] *= dim_perm[i]
+            if dim_perm[i] == -1:
+                test_name += "_negdimarg" + str(arg_idx)
+        new_constructor_args = tuple(new_constructor_args)
```

Comment:
yes,
The first one liner does not work though because not all args are possible dimensions.
I modified it to work, but I am not sure if its clearer than the for loop, let me know.